src/ops/mod.rs

```diff
@@ -531,11 +531,15 @@ impl HttpHandler {
            req_p.extension().and_then(|s| s.to_str()).map(|s| !BLACKLISTED_ENCODING_EXTENSIONS.contains(&UniCase::new(s))).unwrap_or(true) {
             self.handle_get_file_encoded(req, req_p, mime_type)
         } else {
+            let file = match File::open(&req_p) {
+                Ok(file) => file,
+                Err(err) => return self.handle_requested_entity_forbidden(req, err, "file"),
```

Comment:
Are you presuming here that the only reason for an error is that it's forbidden?

Reply:
More that that function only gracefully handles `ErrorKind::Forbidden`, but calling it `handle_requested_entity_open_error()` is more like it, you're right.